src/Hydra/Serializer/CollectionFiltersNormalizer.php

```diff
@@ -159,4 +156,30 @@ private function getSearch(string $resourceClass, array $parts, array $filters):
 
         return ['@type' => 'hydra:IriTemplate', 'hydra:template' => sprintf('%s{?%s}', $parts['path'], implode(',', $variables)), 'hydra:variableRepresentation' => 'BasicRepresentation', 'hydra:mapping' => $mapping];
     }
+
+    private ?ContainerInterface $filterLocator = null;
+
+    /**
+     * Sets a filter locator with a backward compatibility.
```

Comment:
what's the backward compatibility? maybe we can simplify this and actually remove this method?

Reply:
I notice afterwards that I removed the compatibility layer, I will update this to restore the behaviour

Reply:
I might merge both of the trait methods in a single  one